controller/replica_controller.go

```diff
@@ -585,12 +585,14 @@ func (rc *ReplicaController) enqueueNodeChange(obj interface{}) {
 
 	// Add eviction requested replicas to the workqueue
 	for _, diskName := range evictionDisks {
-		for replicaName := range node.Status.DiskStatus[diskName].ScheduledReplica {
-			replica, err := rc.ds.GetReplica(replicaName)
-			if err != nil {
-				return
+		if diskStatus, existed := node.Status.DiskStatus[diskName]; existed {
+			for replicaName := range diskStatus.ScheduledReplica {
+				replica, err := rc.ds.GetReplica(replicaName)
+				if err != nil {
+					return
+				}
+				rc.enqueueReplica(replica)
```

Comment:
```suggestion
				if replica, err := rc.ds.GetReplica(replicaName); err == nil {
					rc.enqueueReplica(replica)
				}
```
If one replica cannot be retrieved from the datastore, for example because it has been deleted in the mean time.
We still want to enqueue the other replicas so they can run a reconcile loop.

Reply:
You can also merge the two loops and get rid of the evictionDisk slice.
```
evictionRequested := node.Spec.EvictionRequested || diskSpec.EvictionRequested
if diskStatus, existed := node.Status.DiskStatus[diskName]; existed && evictionRequested {
	for replicaName := range diskStatus.ScheduledReplica {
		if replica, err := rc.ds.GetReplica(replicaName); err == nil {
			rc.enqueueReplica(replica)
		}
	}
}

}